tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -431,6 +431,38 @@ public void IsNotMacBuild (ApplePlatform platform, string runtimeIdentifiers, bo
 			}
 		}
 
+		[Test]
+		[TestCase (ApplePlatform.MacCatalyst, "maccatalyst-arm64;maccatalyst-x64")]
+		public void IsOverrideRuntimeIdentifier (ApplePlatform platform, string runtimeIdentifiers)
+		{
+			var project = "MySimpleApp";
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+			Configuration.AssertRuntimeIdentifiersAvailable (platform, runtimeIdentifiers);
+			var project_path = GetProjectPath (project, runtimeIdentifiers: runtimeIdentifiers, platform: platform, out var appPath);
+			Clean (project_path);
+			var properties = GetDefaultProperties (runtimeIdentifiers);
+			properties ["RuntimeIdentifier"] = "maccatalyst-arm64";
+			DotNet.AssertBuild (project_path, properties);
```

Comment:
You can get the warnings and assert that we get the expected one:

```suggestion
			var rv = DotNet.AssertBuild (project_path, properties);
			var warnings = BinLog.GetBuildLogWarnings (rv.BinLogPath).ToArray ();
			// the assert here
```